### PR TITLE
TLS: Drop messages associated with a closed endpoint in the processes

### PR DESCRIPTION
**Pull request by jkralik**
When a session has been closed, we also need to clean up the process events associated with the endpoint.

### CODE REVIEW

File: security/oc_tls.c [hunk 1]
```diff
@@ -393,18 +393,32 @@ process_drop_event_for_removed_endpoint(oc_process_event_t ev,
...
   oc_message_t *message = (oc_message_t *)data;
   if (oc_endpoint_compare(&message->endpoint, endpoint) == 0) {
 #if OC_DBG_IS_ENABLED
-    oc_string64_t endpoint_str;
-    oc_endpoint_to_string64(&message->endpoint, &endpoint_str);
-    OC_DBG("oc_tls: dropping %s message for removed endpoint(%s)",
-           (ev == oc_event_to_oc_process_event(RI_TO_TLS_EVENT)) ? "sent"
-                                                                 : "received",
+    oc_string_t endpoint_str;
+    oc_endpoint_to_string(&message->endpoint, &endpoint_str);
```
Comment (Danielius1922) [inline security/oc_tls.c:405]: oc_endpoint_to_string64 to avoid allocation and deallocation

Comment (jkralik) [reply security/oc_tls.c]: fixed

**Comment by jkralik** [comment]
👍

File: port/common/unix/oc_allocator.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "port/oc_allocator_internal.h"
+#include "port/oc_assert.h"
+
+#ifndef OC_DYNAMIC_ALLOCATION
+
+#include <pthread.h>
```
Comment (jkralik) [inline port/common/unix/oc_allocator.c:24]: Unix - i think the posix i better name for directory

**Pull request merged**